File: js/twosum_1.js
const twoSum = (nums, target) => {
  let map = new Map();

  for (let i = 0; i < nums.length; i++) {
    var diff = target - nums[i];

    if (map.has(diff)) {
      return [map.get(diff), i];
    }

    map.set(nums[i], i);
  }

  return map;
};

const resp = twoSum([2, 7, 11, 15], 9);
console.log("resp", resp);

File: dart/two_sum_1.dart
void main() {
  final resp = twoSum([3, 2, 4], 6);
  print(resp);
}

List<int> twoSum(List<int> nums, int target) {
  Map<int, int> map = {};

  for (int i = 0; i < nums.length; i++) {
    int diff = target - nums[i];

    if (map.containsKey(diff)) {
      return [map[diff]!, i];
    } else {
      map[nums[i]] = i;
    }
  }

  return [-1];
}
